### PR TITLE
add multidimentional RCA API

### LINKED ISSUE #16984: [Cognitive Services - Anomaly Detector] Multidimensional Root Cause Analysis
• Key contact for the review:  suxi@microsoft.com
• Whether this is a new or existing API:  This is a set of new API
• Whether or not your service already been deployed:   Hasn't been deployed
• A brief description of the material to be reviewed/change:  This new feature from Anomaly Detector service is Multidimensional Root Cause Analysis (MDRCA). It can discover the root causes of problems to identify appropriate solutions within your complex metric systems such as ROI, CTR, through detecting the abnormal dimension(s) or abnormal dimension combinations when there is a big spike or drop in your metrics system, without knowing any machine learning knowledge, labeled data, or even anomaly information.
• A link to the PR/issue in  azure/azure-rest-api-specs [link](https://github.com/Azure/azure-rest-api-specs/pull/18037)

### PR DESCRIPTION
**Pull request by suxi-ms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Have these APIs been reviewed by the API review board?

**Comment by suxi-ms** [comment]
> Have these APIs been reviewed by the API review board?

Not yet. We are scheduling the review time with review board.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 1]
```diff
@@ -0,0 +1,2031 @@
...
+        "responses": {
+          "201": {
+            "description": "Success",
+            "headers": {
+              "Location": {
+                "description": "Location of the newly created resource.",
+                "type": "string"
+              }
+            }
+          },
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:41]: All success responses except 202 & 204 should define a response body.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 2]
```diff
@@ -0,0 +1,2031 @@
...
+    "MultiDimRCADataMetaListResponse": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "@nextLink": {
+          "description": "Link of next page, only provided when there are next pages",
+          "type": "string",
+          "readOnly": true
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:1476]: There is no need to define `nextLink` as `readOnly: true` as this schema is only used in a response.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 3]
```diff
@@ -0,0 +1,2031 @@
...
+        "parameters": [
+          {
+            "in": "query",
+            "name": "name",
+            "description": "Multidimensional root cause analysis data set name",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "$skip",
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:73]: DO NOT use the "$" prefix on OData query parameter names -- the Azure standard is to drop the "$".

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 4]
```diff
@@ -0,0 +1,2031 @@
...
+            "description": "For paging, skipped number",
+            "type": "integer",
+            "format": "int32"
+          },
+          {
+            "in": "query",
+            "name": "$maxpagesize",
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:80]: Don't use "$" in query parameter names.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 5]
```diff
@@ -0,0 +1,2031 @@
...
+        }
+      }
+    },
+    "MultiDimRCADataSetList": {
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "@nextLink": {
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:1060]: Don't use the "@" in the name of the `nextLink` property of pageable responses.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 6]
```diff
@@ -0,0 +1,2031 @@
...
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "body",
+            "name": "body",
+            "description": "create a multidimensional root cause analysis data set request",
+            "schema": {
+              "$ref": "#/definitions/CreateMultiDimRCADatasetRequest"
+            }
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:29]: Should this body parameter be marked as required?  The body schema has required fields so it seems unlikely that the entire body is optional.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 7]
```diff
@@ -0,0 +1,2031 @@
...
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Data to be uploaded to perform multidimensional root cause analysis",
+            "schema": {
+              "$ref": "#/definitions/MultiDimRCAUploadDataRequest"
+            }
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:215]: Should body parameter be required?

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 8]
```diff
@@ -0,0 +1,2031 @@
...
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Create a multidimensional RCA instance request",
+            "schema": {
+              "$ref": "#/definitions/MultiDimRCAInstanceRequest"
+            }
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:394]: Should body parameter be required?

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 9]
```diff
@@ -0,0 +1,2031 @@
...
+            "format": "uuid"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Training request of multidimensional RCA",
+            "schema": {
+              "$ref": "#/definitions/MultiDimRCATrainingRequest"
+            }
+          }
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:581]: Should body parameter be required?

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json [hunk 10]
```diff
@@ -0,0 +1,2031 @@
...
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Inference request of multidimensional RCA",
+            "schema": {
+              "$ref": "#/definitions/MultiDimRCAInferenceRequest"
+            }
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v2.0-preview/MultidimensionalRootCauseAnalysis.json:798]: Should body parameter be required?

**Review by mikekistler — COMMENTED** [review]
I made a quick pass and found a few issues.

**Pull request closed without merging**